### PR TITLE
methods for shared secret

### PR DESCRIPTION
**Pull request by isentropy**
It's working in my local tests, but I had trouble adding integration test: it seems that api won't add member using secret to a DU that was created outside of E&E. Is that right @jtakalai?

### CODE REVIEW

**Comment by isentropy** [comment]
also added: connection time passed to OkHttpClient

File: src/main/java/com/streamr/client/StreamrRESTClient.java [hunk 1]
```diff
@@ -65,9 +69,18 @@ public StreamrRESTClient(StreamrClientOptions options) {
         }
     }
 
-    private <T> T execute(Request request, JsonAdapter<T> adapter) throws IOException {
-        OkHttpClient client = new OkHttpClient();
+    protected OkHttpClient.Builder httpClientBuilder(){
```
Comment (isentropy) [inline src/main/java/com/streamr/client/StreamrRESTClient.java:72]: @kare Config timeout is passed to OkHttpClient here

**Review by jtakalai — APPROVED** [review]
LGTM

Product class is marked "just for testing", should it be package-private instead of public or something? Thinking, would someone try to use it... but for now even that is totally a non-issue for sure.

**Comment by isentropy** [comment]
> LGTM
> 
> Product class is marked "just for testing", should it be package-private instead of public or something? Thinking, would someone try to use it... but for now even that is totally a non-issue for sure.

y package private is better. I made the change

**Pull request merged**